src/main/java/github/scarsz/discordsrv/listeners/DiscordChatListener.java

```diff
@@ -59,9 +62,16 @@
     @Override
     public void onGuildMessageReceived(GuildMessageReceivedEvent event) {
         // if message is from null author or self do not process
-        if (event.getMember() == null || DiscordUtil.getJda() == null || event.getAuthor().equals(DiscordUtil.getJda().getSelfUser()))
+        if ((event.getMember() == null && !event.isWebhookMessage()) || DiscordUtil.getJda() == null || event.getAuthor().equals(DiscordUtil.getJda().getSelfUser()))
             return;
 
+        Webhook webhook = event.getChannel().retrieveWebhooks().complete().stream()
+                .filter(hook -> hook.getName().equals("DiscordSRV " + event.getChannel().getId()))
+                .findFirst().orElse(null);
```

Comment:
This request should be avoided by caching the ID instead of **getting all webhooks every time there is a Discord message**

Reply:
Without a reliable way to update the cache of the current webhooks in a channel, I instead made this part of the code only run if it is a webhook message and moved the `BlockWebhooks` check above this part to reduce the amount of times the webhooks are retrieved.

Reply:
This still needs to be resolved, technically the webhook id is already cached, since the webhook url is cached and the id is a part of it. You could even add caching for the webhook id along side the webhook url caching that's present right now

Reply:
I've used the WebhookUtil.getWebhookUrlToUseForChannel() method instead to hopefully resolve this.